### PR TITLE
samba4: mark enableGlusterFS as broken

### PR DESCRIPTION
**Pull request by lovesegfault**
###### Motivation for this change
I noticed this was broken while testing #112178

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/samba/4.x.nix [hunk 1]
```diff
@@ -158,7 +158,7 @@ stdenv.mkDerivation rec {
     description = "The standard Windows interoperability suite of programs for Linux and Unix";
     license = licenses.gpl3;
     platforms = platforms.unix;
-    broken = stdenv.isDarwin;
+    broken = stdenv.isDarwin || enableGlusterFS;
```
Comment (cole-h) [inline pkgs/servers/samba/4.x.nix:161]: Maybe add a comment *why* this is broken when used in conjunction with `enableGlusterFS`?

Also I'd reword the commit to be `samba4: mark broken when used with enableGlusterFS option` or something.

Comment (lovesegfault) [reply pkgs/servers/samba/4.x.nix:161]: The build failing is the reason btw

Comment (lovesegfault) [reply pkgs/servers/samba/4.x.nix:162]: Alright, added a comment and changed the commit message

**Review by cole-h — APPROVED** [review]
Diff LGTM.

**Comment by mweinelt** [comment]
This change broke sambaFull, because it sets `enableGlusterFS = true;`.

**Comment by lovesegfault** [comment]
> This change broke sambaFull, because it sets `enableGlusterFS = true;`.

IIRC sambaFull was already broken, because enableGlusterGS was broken. This just marked it as broken.

Did the feature compile in your experience?

**Comment by mweinelt** [comment]
No, it did not. But unbreaking the samba package by disabling glusterfs would've been the right move IMO.

**Comment by lovesegfault** [comment]
> No, it did not. But unbreaking the samba package by disabling glusterfs would've been the right move IMO.

That would mean silently breaking the setup of anyone who had `sambaFull` and relied on `enableGlusterFS` whenever they updated. 

This is in no small part why I dislike these `Full` packages. With them we basically can't deprecate `enableFoo` and have it warn users who had manually enabled that, without breaking the setup of users who maybe didn't even know that `enableFoo` was a thing.

I suspect the right thing to do here is to ask around (matrix, discourse, issues) if anyone out there cares to maintain samba's glusterfs feature and if not we can just drop it.

**Comment by andir** [comment]
You could a) mention it in the release notes and b) throw an assertion if that broken feature flag is passed. This will not solve the problem but warn users about it. We can't keep the same features in Full packages all the time. At least not across releases. Otherwise we will have to call packages SambaWithAAndBAndCAnd...

**Comment by mweinelt** [comment]
Releasing sambaFull in a broken state into 21.05 is just the wrong move altogether, no matter the solution we ultimately pick.

**Comment by lovesegfault** [comment]
> Releasing sambaFull in a broken state into 21.05 is just the wrong move altogether, no matter the solution we ultimately pick.

If I fix this today, would you do the backport to 21.05?

**Comment by mweinelt** [comment]
Already fixed in #125105, I'm working on the backport now.

**Comment by lovesegfault** [comment]
Thanks! Sorry for the breakage!
